src/api/app/models/kiwi/image.rb

```diff
@@ -48,43 +49,51 @@ def self.build_from_xml(xml_string, md5)
     order = 1
     repositories = xml["repository"]
     repositories = [xml["repository"]] if xml["repository"].is_a?(Hash)
-    repositories.each do |repository|
-      attributes = {
-        repo_type:   repository['type'],
-        source_path: repository['source']['path'],
-        priority:    repository['priority'],
-        order:       order,
-        alias:       repository['alias'],
-        replaceable: repository['status'] == 'replaceable',
-        username:    repository['username'],
-        password:    repository['password']
-      }
-      attributes['imageinclude'] = repository['imageinclude'] == 'true' if repository.key?('imageinclude')
-      attributes['prefer_license'] = repository['prefer-license'] == 'true' if repository.key?('prefer-license')
-
-      new_image.repositories.build(attributes)
-      order += 1
+
+    if repositories
+      new_image.use_project_repositories = true if repositories.map { |repository| repository['source']['path'] }.include?('obsrepositories:/')
+
+      repositories.reject{ |repository| repository['source']['path'] == 'obsrepositories:/' }.each do |repository|
```

Comment:
``repositories.reject{ |repository| repository['source']['path'] == 'obsrepositories:/' }.each.with_index(1) do |repository, index|``

and you can get rid of the order variable.